src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/TalismanListener.java

```diff
@@ -239,7 +226,15 @@ public void onEnchant(EnchantItemEvent e) {
             TalismanEnchantment enchantment = talisman.getRandomEnchantment(e.getItem(), enchantments.keySet());
 
             if (enchantment != null) {
-                enchantments.put(enchantment.getEnchantment(), enchantment.getLevel());
+                /*
+                 * Fix #2679
+                 * By default, the Bukkit API doesn't allow us to give enchantment books extra enchantments.
+                 */
+                if (talisman.isEnchantmentBookAllowed() && e.getItem().getType() == Material.BOOK) {
```

Comment:
The issue with #2679 included that the talisman will send a message even if no enchantment was added.
This doesn't really fix this issue because the message is sent in line 224 in the `Talisman.checkFor` method, so to prevent the message from being sent, we would have to move this check before that.